packages/typespec-azure-resource-manager/src/rules/arm-resource-provisioning-state-rule.ts

```diff
@@ -9,7 +10,7 @@ export const armResourceProvisioningStateRule = createRule({
   description: "Check for properly configured provisioningState property.",
   messages: {
     default:
-      "The RP-specific property model in the 'properties' property of this resource must contain a 'provisioningState property.  The property type should be an enum, and it must specify known state values 'Succeeded', 'Failed', and 'Canceled'.",
+      "The RP-specific property model in the 'properties' property of this resource must contain a 'provisioningState property.  The property type should be an enum or a union of string values, and it must specify known state values 'Succeeded', 'Failed', and 'Canceled'.",
     missingValues: paramMessage`The "@knownValues" decorator for provisioningState, must reference an enum with 'Succeeded', 'Failed', 'Canceled' values. The enum is missing the values: [${"missingValues"}].`,
```

Comment:
nit: Can you add recommend spreading `ResoourceProvisioningState` rather than declare themselves?

Reply:
well we can't spread with union we just include the enum as a variant. That message is already quite long I think we should have the markdown doc written for it though to show all the details.

We might also want to chagne this rule when we enabvle the `no-enum` rule to just say use union